mindsdb_native/libs/backends/lightwood.py

```diff
@@ -22,88 +25,88 @@ def __init__(self, transaction):
         self.nr_predictions = self.transaction.lmd['tss']['nr_predictions']
         self.nn_mixer_only = False
 
-    def _create_timeseries_df(self, original_df):
-        group_by = self.transaction.lmd['tss']['group_by'] if self.transaction.lmd['tss']['group_by'] is not None else []
-        order_by = self.transaction.lmd['tss']['order_by']
+    def _ts_reshape(self, original_df):
+        gb_arr = self.transaction.lmd['tss']['group_by'] if self.transaction.lmd['tss']['group_by'] is not None else []
+        ob_arr = self.transaction.lmd['tss']['order_by']
         window = self.transaction.lmd['tss']['window']
 
+        original_index_list = []
+        idx = 0
+        for _, row in original_df.iterrows():
```

Comment:
If I'm not mistaken, `df.itertuples()` would be much faster than `df.iterrows()`. Something like:

```
for row in original_df.itertuples():
    ...
```

and 

```
def _make_pred(row):
    return not hasattr(row, "make_predictions") or row.make_predictions
```

should work.

Reply:
[Source](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.iterrows.html) for `itertuples()` faster than `iterrows()` claim. [Here's](https://github.com/dimgold/Datathon-TAU/blob/master/7.6%20Pandas%20iterations%20test/Pandas_Iterations.ipynb) a very simple test, and in [this](https://stackoverflow.com/questions/24870953/does-pandas-iterrows-have-performance-issues) Q&A the graphs at the bottom seem to support this.